client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetIndexResponseTests.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indices;
+
+import org.apache.lucene.util.CollectionUtil;
+import org.elasticsearch.client.GetAliasesResponseTests;
+import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.elasticsearch.cluster.metadata.MappingMetaData;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.settings.IndexScopedSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.RandomCreateIndexGenerator;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class GetIndexResponseTests extends ESTestCase {
+
+    private static final int NUMBER_OF_RUNS = 20;
+
+    /**
+     * check that we can parse the xContent output of a client side response and get the same information back
+     *
+     * TODO: ideally we would create a random instance of the client side response that contains the rendering code
+     * write it to xContent, parse it back and compare the results
+     * this currently would mean we would need test dependencies to server:test
```

Comment:
I'm not sure I really understand this TODO. For what purpose would we need a dependency on `server:test`?

Reply:
An update: as I was working on the same refactor for 'get mappings', I found it nice to use an `xContentTester`. I ended up creating a method `GetMappingsResponseTests#toXContent` that converts an HLRC `GetMappingsResponse` to a server-side one, then calls `toXContent` on that. Here's the test class in case you'd like to take a look: https://github.com/elastic/elasticsearch/pull/37796/files#diff-f66406b858061334c3239c8d2148b2caR41

Reply:
Thanks, I'll take a look...

Reply:
I see what you did there, I think its basically the same as my approach, although I have to say I really dislike using  xContentTester for all the fuss around having to set up functions for everything. If you've used it several times it becomes clearer what it does, but its over-engineered from my point of view. I can change to using it if you really like though.
The main point though is to get an xContent rendering of the server side response. While you are generating a random instance of the client-side class, then convert it to server-side to call its xContent, I copied the code to create a randomized server-side response directly from `org.elasticsearch.action.admin.indices.get.GetIndexResponseTests`. I think it boild down to the same disadvantage, namely having to copy randomization code (which you also seem to have copied over, at least partially). By my comment I mean it would be great if we could call `org.elasticsearch.action.admin.indices.get.GetIndexResponseTests#createTestInstance` in the client-side test to be sure we get a "real" server side response with all its randomization variants, and call its "real" toXContent, because that is what the client is going to have to parse. In order to avoid code changing on the server side unnoticed it would be good to actually use that code, not rely on the copy being up to date which we have to do if we copy. Does this make sense?

The problem I'm running into here is that the client project can see `org.elasticsearch.action.admin.indices.get.GetIndexResponse` but not `org.elasticsearch.action.admin.indices.get.GetIndexResponseTests.` Maybe we could have a test dependency for this? But @hub-cap might know. We can probably do this in a follow up though if people think its useful to have.

Reply:
Thanks, I understand your comment now! Doing a bit more digging, I see `AbstractHlrcXContentTestCase`, which I think was designed for this situation. This sort of class tests both server serialization and HLRC serialization at the same time, which means there is a single place to generate random server objects. HLRC serialization is tested by generating a random server object, calling `toXContent` on it, and parsing it back as an client object. However, this class currently lives in the x-pack directory, which doesn't seem like a natural place to move `GetMappingsResponseTests`, `GetIndexResponseTests`, etc.

A proposal for what to do with our two open PRs:
- We can plan to follow-up with @hub-cap to discuss where to move `AbstractHlrcXContentTestCase`, then refactor `Get*Tests` to all extend this class.
- For now we can generate random server objects (as you do in this PR). I'll switch my PR to do that instead of generating a client-side one, so we are closer to modeling what actually happens during an HLRC call.
- I would prefer us to use `xContentTester`, because although it's a bit overkill, it is used extensively and I think being consistent with the test framework leads to overall less confusion than being inconsistent but slightly simpler in one or two tests.